Bug:[Chassis] platform_tests/api testcases error out in setup

### Is it platform specific

generic

### Importance or Severity

Critical

### Description of the bug

This issue is caused by PR #20942 (Update platform api server to support IPv6)
I have added comment to PR about this issue.

This PR is causing all platform_tests/api (150 testcases) to error out in setup.
This PR will not work for a chassis with multiple hardware SKUs.

Example:
test_get_name runs on board202 and cpm and both error out with the same message.
platform_tests/api/test_chassis.py::TestChassisApi::test_get_name[ixre-egl-board202]::setup
platform_tests/api/test_chassis.py::TestChassisApi::test_get_name[ixre-cpm-chassis20]::setup

```
>           res = localhost.wait_for(host=dut_ip, port=SERVER_PORT, state='started', delay=1, timeout=10)
common/platform/device_utils.py:1133: 

        if (res.is_failed or 'exception' in res) and not module_ignore_errors:
>           raise RunAnsibleModuleFail("run module {} failed".format(self.module_name), res)
E           tests.common.errors.RunAnsibleModuleFail: run module wait_for failed, Ansible Results =>
E           failed = True
E           elapsed = 10
E           msg = Timeout when waiting for 100.117.29.23:8000
E           invocation = {'module_args': {'host': '100.117.29.23', 'port': 8000, 'state': 'started', 'delay': 1, 'timeout': 10, 'connect_timeout': 5, 'active_connection_states': ['ESTABLISHED', 'FIN_WAIT1', 'FIN_WAIT2', 'SYN_RECV', 'SYN_SENT', 'TIME_WAIT'], 'sleep': 1, 'path': None, 'search_regex': None, 'exclude_hosts': None, 'msg': None}}
E           _ansible_no_log = None
E           changed = False
E           stdout =
E           stderr =
common/devices/base.py:134: RunAnsibleModuleFail
```

since fixture duthost_mgmt_ip is called only from start_platform_api_service, all existing cards in the chassis 
will have the address of the first card in the chassis.
In the above example address 100.117.29.23 is the address for board201 in the chassis.  However, when the tests run on board202 or CPM, both use the address of board201 to connect which will cause failure.

### Steps to Reproduce

run the testcase below or any testcase in platform_tests/api.
platform_tests/api/test_chassis.py::TestChassisApi::test_get_name

### Actual Behavior and Expected Behavior

Actual: 
Errors in setup

Exected:
The testcase to run.

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/common/devices/sonic.py
+++ b/tests/common/devices/sonic.py
@@ -3340,6 +3340,28 @@ print(device_prefix)
 
         logging.info("Successfully cleaned up all console sessions")
 
+    def get_mgmt_ip(self):
+        """
+        Gets the management IP address (v4 or v6) on eth0.
+        Defaults to IPv4 on a dual stack configuration.
+        """
+        ipv4_regex = re.compile(r"(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/\d+")
+        ipv6_regex = re.compile(r"([a-fA-F0-9:]+)/\d+")
+
+        mgmt_interface = self.shell("show ip interface | egrep '^eth0 '", module_ignore_errors=True)["stdout"]
+        if mgmt_interface:
+            match = ipv4_regex.search(mgmt_interface)
+            if match:
+                return {"mgmt_ip": match.group(1), "version": "v4"}
+
+        mgmt_interface = self.shell("show ipv6 interface | egrep '^eth0 '", module_ignore_errors=True)["stdout"]
+        if mgmt_interface:
+            match = ipv6_regex.search(mgmt_interface)
+            if match:
+                return {"mgmt_ip": match.group(1), "version": "v6"}
+
+        assert False, "Failed to find duthost mgmt ip"  # noqa: F631
+
 
 def assert_exit_non_zero(shell_output):
     if shell_output['rc'] != 0:

--- a/tests/common/fixtures/duthost_utils.py
+++ b/tests/common/fixtures/duthost_utils.py
@@ -8,7 +8,6 @@ import collections
 import ipaddress
 import time
 import json
-import re
 
 from pytest_ansible.errors import AnsibleConnectionFailure
 from paramiko.ssh_exception import AuthenticationException
@@ -939,31 +938,6 @@ def duthosts_ipv6_mgmt_only(duthosts, backup_and_restore_config_db_on_duts):
                               cmd_desc="netstat")
 
     return duthosts
-
-
-@pytest.fixture(scope="module")
-def duthost_mgmt_ip(duthosts, enum_rand_one_per_hwsku_hostname):
-    """
-    Gets the management IP address (v4 or v6) on eth0.
-    Defaults to IPv4 on a dual stack configuration.
-    """
-    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
-    ipv4_regex = re.compile(r"(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})/\d+")
-    ipv6_regex = re.compile(r"([a-fA-F0-9:]+)/\d+")
-
-    mgmt_interface = duthost.shell("show ip interface | egrep '^eth0 '", module_ignore_errors=True)["stdout"]
-    if mgmt_interface:
-        match = ipv4_regex.search(mgmt_interface)
-        if match:
-            return {"mgmt_ip": match.group(1), "version": "v4"}
-
-    mgmt_interface = duthost.shell("show ipv6 interface | egrep '^eth0 '", module_ignore_errors=True)["stdout"]
-    if mgmt_interface:
-        match = ipv6_regex.search(mgmt_interface)
-        if match:
-            return {"mgmt_ip": match.group(1), "version": "v6"}
-
-    pt_assert(False, "Failed to find duthost mgmt ip")
 
 
 def assert_addr_in_output(addr_set: Dict[str, List], hostname: str,

--- a/tests/common/helpers/gnmi_utils.py
+++ b/tests/common/helpers/gnmi_utils.py
@@ -404,14 +404,12 @@ def verify_tcp_port(localhost, ip, port):
     logger.info("TCP: " + res['stdout'] + res['stderr'])
 
 
-def gnmi_capabilities(duthost, localhost, duthost_mgmt_ip=None):
+def gnmi_capabilities(duthost, localhost):
     env = GNMIEnvironment(duthost, GNMIEnvironment.GNMI_MODE)
-    if duthost_mgmt_ip:
-        ip = duthost_mgmt_ip['mgmt_ip']
-        addr = f"[{ip}]" if duthost_mgmt_ip['version'] == 'v6' else f"{ip}"
-    else:
-        ip = duthost.mgmt_ip
-        addr = ip
+    duthost_mgmt_info = duthost.get_mgmt_ip()
+    ip = duthost_mgmt_info['mgmt_ip']
+    addr = f"[{ip}]" if duthost_mgmt_info['version'] == 'v6' else f"{ip}"
+
     port = env.gnmi_port
     # Run gnmi_cli in gnmi container as workaround
     cmd = "docker exec %s gnmi_cli -client_types=gnmi -a %s:%s " % (env.gnmi_container, addr, port)

--- a/tests/common/platform/device_utils.py
+++ b/tests/common/platform/device_utils.py
@@ -20,7 +20,6 @@ from tests.common.platform.reboot_timing_constants import SERVICE_PATTERNS, OTHE
     OFFSET_ITEMS, TIME_SPAN_ITEMS, REQUIRED_PATTERNS
 from tests.common.devices.duthosts import DutHosts
 from tests.common.plugins.ansible_fixtures import ansible_adhoc  # noqa: F401
-from tests.common.fixtures.duthost_utils import duthost_mgmt_ip  # noqa: F401
 
 """
 Helper script for fanout switch operations
@@ -1081,10 +1080,12 @@ def advanceboot_neighbor_restore(duthosts, enum_rand_one_per_hwsku_frontend_host
 
 
 @pytest.fixture(scope='function')
-def start_platform_api_service(duthosts, enum_rand_one_per_hwsku_hostname, duthost_mgmt_ip,  # noqa: F811
+def start_platform_api_service(duthosts, enum_rand_one_per_hwsku_hostname,
                                localhost, request):
     duthost = duthosts[enum_rand_one_per_hwsku_hostname]
-    dut_ip = duthost_mgmt_ip['mgmt_ip']
+    duthost_mgmt_info = duthost.get_mgmt_ip()
+    dut_ip = duthost_mgmt_info['mgmt_ip']
+    dut_mgmt_ver = duthost_mgmt_info['version']
 
     res = localhost.wait_for(host=dut_ip,
                              port=SERVER_PORT,
@@ -1102,7 +1103,7 @@ def start_platform_api_service(duthosts, enum_rand_one_per_hwsku_hostname, dutho
             'command=/usr/bin/python{} /opt/platform_api_server.py --port {} {}'.format(
                 '3' if py3_platform_api_available else '2',
                 SERVER_PORT,
-                '--ipv6' if duthost_mgmt_ip['version'] == 'v6' else ''),
+                '--ipv6' if dut_mgmt_ver == 'v6' else ''),
             'autostart=True',
             'autorestart=True',
             'stdout_logfile=syslog',
@@ -1121,7 +1122,7 @@ def start_platform_api_service(duthosts, enum_rand_one_per_hwsku_hostname, dutho
         duthost.command('docker cp {} pmon:{}'.format(dest_path, pmon_path))
 
         # Prepend an iptables rule to allow incoming traffic to the HTTP server
-        if duthost_mgmt_ip['version'] == 'v6':
+        if dut_mgmt_ver == 'v6':
             duthost.command(IP6TABLES_PREPEND_RULE_CMD)
         else:
             duthost.command(IPTABLES_PREPEND_RULE_CMD)
@@ -1135,8 +1136,9 @@ def start_platform_api_service(duthosts, enum_rand_one_per_hwsku_hostname, dutho
 
 
 @pytest.fixture(scope='function')
-def platform_api_conn(duthost_mgmt_ip, start_platform_api_service):  # noqa: F811
-    dut_ip = duthost_mgmt_ip['mgmt_ip']
+def platform_api_conn(duthosts, enum_rand_one_per_hwsku_hostname, start_platform_api_service):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    dut_ip = duthost.get_mgmt_ip()["mgmt_ip"]
 
     conn = http.client.HTTPConnection(dut_ip, 8000)
     try:

--- a/tests/gnmi/test_gnmi.py
+++ b/tests/gnmi/test_gnmi.py
@@ -6,7 +6,6 @@ from tests.common.helpers.gnmi_utils import gnmi_capabilities, add_gnmi_client_c
 from .helper import gnmi_set, dump_gnmi_log
 from tests.common.utilities import wait_until
 from tests.common.plugins.allure_wrapper import allure_step_wrapper as allure
-from tests.common.fixtures.duthost_utils import duthost_mgmt_ip       # noqa: F401
 
 
 logger = logging.getLogger(__name__)
@@ -18,12 +17,12 @@ pytestmark = [
 ]
 
 
-def test_gnmi_capabilities(duthosts, rand_one_dut_hostname, localhost, duthost_mgmt_ip):  # noqa: F811
+def test_gnmi_capabilities(duthosts, rand_one_dut_hostname, localhost):
     '''
     Verify GNMI capabilities
     '''
     duthost = duthosts[rand_one_dut_hostname]
-    ret, msg = gnmi_capabilities(duthost, localhost, duthost_mgmt_ip)
+    ret, msg = gnmi_capabilities(duthost, localhost)
     assert ret == 0, (
         "GNMI capabilities command failed (non-zero return code).\n"
         "- Error message: {}"
@@ -40,7 +39,7 @@ def test_gnmi_capabilities(duthosts, rand_one_dut_hostname, localhost, duthost_m
     ).format(msg)
 
 
-def test_gnmi_capabilities_authenticate(duthosts, rand_one_dut_hostname, localhost, duthost_mgmt_ip):  # noqa: F811
+def test_gnmi_capabilities_authenticate(duthosts, rand_one_dut_hostname, localhost):
     '''
     Verify GNMI capabilities with different roles
     '''
@@ -49,7 +48,7 @@ def test_gnmi_capabilities_authenticate(duthosts, rand_one_dut_hostname, localho
     with allure.step("Verify GNMI capabilities with noaccess role"):
         role = "gnmi_noaccess"
         add_gnmi_client_common_name(duthost, "test.client.gnmi.sonic", role)
-        ret, msg = gnmi_capabilities(duthost, localhost, duthost_mgmt_ip)
+        ret, msg = gnmi_capabilities(duthost, localhost)
         assert ret != 0, (
             "GNMI capabilities authenticate with noaccess role command unexpectedly succeeded "
             "(zero return code) for a client with noaccess role.\n"
@@ -63,7 +62,7 @@ def test_gnmi_capabilities_authenticate(duthosts, rand_one_dut_hostname, localho
     with allure.step("Verify GNMI capabilities with readonly role"):
         role = "gnmi_readonly"
         add_gnmi_client_common_name(duthost, "test.client.gnmi.sonic", role)
-        ret, msg = gnmi_capabilities(duthost, localhost, duthost_mgmt_ip)
+        ret, msg = gnmi_capabilities(duthost, localhost)
         assert ret == 0, (
             "GNMI capabilities authenticate readonly command failed (non-zero return code).\n"
             "- Error message: {}"
@@ -80,7 +79,7 @@ def test_gnmi_capabilities_authenticate(duthosts, rand_one_dut_hostname, localho
     with allure.step("Verify GNMI capabilities with readwrite role"):
         role = "gnmi_readwrite"
         add_gnmi_client_common_name(duthost, "test.client.gnmi.sonic", role)
-        ret, msg = gnmi_capabilities(duthost, localhost, duthost_mgmt_ip)
+        ret, msg = gnmi_capabilities(duthost, localhost)
         assert ret == 0, (
             "GNMI capabilities authenticate readwrite role command failed (non-zero return code).\n"
             "- Error message: {}"
@@ -97,7 +96,7 @@ def test_gnmi_capabilities_authenticate(duthosts, rand_one_dut_hostname, localho
     with allure.step("Verify GNMI capabilities with empty role"):
         role = ""
         add_gnmi_client_common_name(duthost, "test.client.gnmi.sonic", role)
-        ret, msg = gnmi_capabilities(duthost, localhost, duthost_mgmt_ip)
+        ret, msg = gnmi_capabilities(duthost, localhost)
         assert ret == 0, (
             "GNMI capabilities authenticate with empty role command failed (non-zero return code).\n"
             "- Error message: {}"

--- a/tests/gnmi/test_mimic_hwproxy_cert_rotation.py
+++ b/tests/gnmi/test_mimic_hwproxy_cert_rotation.py
@@ -6,7 +6,6 @@ from tests.common.helpers.assertions import pytest_assert
 from tests.common.utilities import wait_until
 from tests.common.helpers.gnmi_utils import GNMIEnvironment, gnmi_capabilities
 from tests.common.utilities import get_image_type
-from tests.common.fixtures.duthost_utils import duthost_mgmt_ip      # noqa: F401
 
 
 logger = logging.getLogger(__name__)
@@ -30,8 +29,7 @@ def check_telemetry_status(duthost):
     return "RUNNING" in output['stdout']
 
 
-def test_mimic_hwproxy_cert_rotation(duthosts, rand_one_dut_hostname, localhost, ptfhost,
-                                     duthost_mgmt_ip):  # noqa: F811
+def test_mimic_hwproxy_cert_rotation(duthosts, rand_one_dut_hostname, localhost, ptfhost):
     duthost = duthosts[rand_one_dut_hostname]
 
     # Use bash -c to run the pipeline properly
@@ -89,7 +87,7 @@ def test_mimic_hwproxy_cert_rotation(duthosts, rand_one_dut_hostname, localhost,
             enable_feature = 'sudo config feature state gnmi enabled'
             duthost.command(enable_feature, module_ignore_errors=True)
             assert wait_until(60, 3, 0, check_gnmi_status, duthost), "GNMI service failed to start"
-            ret, msg = gnmi_capabilities(duthost, localhost, duthost_mgmt_ip)
+            ret, msg = gnmi_capabilities(duthost, localhost)
             assert ret == 0, msg
             assert "sonic-db" in msg, msg
             assert "JSON_IETF" in msg, msg

